### PR TITLE
Avoid full-history copies when sampling Chronos-2 context windows

### PR DESCRIPTION
**Pull request by ayoubchikri**
## Problem and change

`Chronos2Dataset._construct_slice` clones the entire prepared history before selecting a context window. A 1,024-step sample from a million-step history therefore copies all million steps and returns a view retaining that complete allocation. This cost recurs for every sampled training window, as well as validation and prediction samples.

Slice first, then clone only the selected context. Keep the existing independent target copy before masking covariates, and clone prepared future covariates only in TEST mode, where they are used.

This keeps returned tensors independent of prepared inputs and preserves sample values, NaN masks, and shapes. The context copy now scales with the selected window rather than the full history.

## Measured impact

Local microbenchmark of `_construct_slice`, using 4 float32 variates, context length 1,024, prediction length 64, and one PyTorch CPU thread:

| History length | Context allocation before | After | TRAIN sample time before | After |
| --- | ---: | ---: | ---: | ---: |
| 10,000 | 160,000 B | 16,384 B | 87.4 µs | 65.9 µs |
| 100,000 | 1,600,000 B | 16,384 B | 207.6 µs | 66.7 µs |
| 1,000,000 | 16,000,000 B | 16,384 B | 5,970.1 µs | 83.6 µs |

Timings are medians of 20 rounds of 10 calls after warmup, interleaving the original and modified methods. These are local sample-construction measurements, not end-to-end model speedups or total process memory. The original prepared history remains allocated. Outputs were compared exactly (including NaNs) in TRAIN, VALIDATION and TEST modes at all three history lengths.

The allocation difference can be reproduced on either revision with:

```python
import numpy as np
import torch
from chronos.chronos2.dataset import Chronos2Dataset, DatasetMode

prepared = {
    "context": torch.arange(4_000_000, dtype=torch.float32).reshape(4, -1),
    "future_covariates": torch.full((4, 64), torch.nan),
    "n_targets": 2,
    "n_covariates": 2,
    "n_future_covariates": 1,
}
ds = Chronos2Dataset(
    [prepared], context_length=1024, prediction_length=64,
    batch_size=4, output_patch_size=16, mode=DatasetMode.TRAIN,
)
np.random.seed(0)
context = ds._construct_slice(0)[0]
print(context.shape, context.untyped_storage().nbytes())
# Before: torch.Size([4, 1024]) 16000000
# After:  torch.Size([4, 1024]) 16384
```

## Validation

- Six local regression checks cover all modes with short and long histories, the underlying storage allocation, exact sample values, target/covariate masks, and mutation isolation. Five fail on upstream `4dbf163`; all six pass after the fix.
- Existing preprocessing and Chronos-2 tests plus the six local regression checks: **333 passed**. The remaining 33 failures have identical test IDs on upstream: five require uncached Hugging Face datasets, and 28 use fused AdamW unsupported by CPU PyTorch 2.2.
- Rerunning with `optim="adamw_torch"` through `fit`'s existing trainer kwargs, excluding external downloads and LoRA, gives **361 passed, 13 deselected**, including actual fine-tuning and prediction tests. This optimizer override was local validation only.
- The changed source passes lint with the pre-existing UP035 import warning excluded. `git diff --check` passes.

Environment: macOS x86_64, Python 3.12.12, PyTorch 2.2.2, Transformers 4.49.0. GPU execution was not tested.

### CODE REVIEW

File: test/test_dataset_window_memory.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (shchur) [inline test/test_dataset_window_memory.py]: I think we can remove these tests since the change has a small and well-defined score.

**Review by shchur — APPROVED** [review]
Thanks for the great contribution! I think we can merge after removing `test/test_dataset_window_memory.py`.
